backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -796,10 +799,9 @@ public Mono<Void> createWorkQueue(QueueSpecification queueSpecification) {
     @Override
     public void close() {
         sender.close();
-        ImmutableList.copyOf(refs.values())
-            .stream()
-            .map(PooledRef::poolable)
-            .forEach(channel -> getChannelCloseHandler().accept(SignalType.ON_NEXT, channel));
+       Flux.fromIterable(refs.values())
+           .flatMap(PooledRef::invalidate)
+           .blockLast();
```

Comment:
Should we use .subscribe(); ?

Reply:
No here we want to be sure that it is finished.

Also stopping the app is not a reactive pipeline: blocking is OK.